IPNI_HTTP_PROVIDER.md

```diff
@@ -0,0 +1,332 @@
+encoding based on specific requirements or preferences. The choice of encoding should be communicated through `Accept`
+and `Content-Type` headers in the HTTP request and response headers. Alternative examples
+include [`application/vnd.ipld.dag-cbor`](https://www.iana.org/assignments/media-types/application/vnd.ipld.dag-cbor).
+
+As a special case, the following content types are interpreted as their IPLD equivelant:
+
+* `application/json` is considered to be the same as `application/vnd.ipld.dag-json`.
+* `application/cbor` is considered to be the same as `application/vnd.ipld.dag-cbor`.
+
+The common data types, such as `CID` and `Multihash`, are encoded in their
+standard [multibase format](https://github.com/multiformats/multibase) when referenced in URLs.
+
+### Versioning
+
+The IPNI Provider HTTP API adopts a versioning strategy that utilizes the path prefix to indicate major releases. As a
+best practice, future iterations that introduce breaking changes should increment the version number in the API path.
+This approach ensures compatibility and allows clients to adapt to new versions while maintaining backward compatibility
+with existing implementations. By explicitly incorporating versioning in the API path, it becomes easier to manage and
+track changes over time, facilitating smooth transitions and providing a clear indication of API evolution.
+
+
+### Namespacing
+
+This specification introduces the `/ipni/` URL prefix to namespace the functionality associated with IPNI HTTP providers. The presence of this namespace allows IPNI specifications to evolve without conflicting with other URL patterns that an HTTP server may support. Therefore, this specification reserves the right for any sub-tree that appears after this prefix.
+
+#### Implicit Namespace Assumption
+
+Indexer nodes implicitly assume the `/ipni/` URL prefix. When receiving an announcement from providers, an indexer will attempt to fetch advertisements by joining the announced URL with the paths documented in this specification. Implementers have the freedom to include additional paths to the announced URL if desired.
+
+For example, if the announced URL is `https://ipni-provider.example`, it will be accessed at `https://ipni-provider.example/ipni/v1/ad/head` to fetch the head advertisement. Similarly, if the announced URL is `https://ipni-provider.example/my/prefix`, it will be accessed at `https://ipni-provider.example/my/prefix/ipni/v1/ad/head`.
+
+#### Custom Paths in Multiaddr
+
+Custom paths in addresses specified as Multiaddr is not currently supported. See related issues below:
+* https://github.com/ipni/go-libipni/issues/42
+* https://github.com/libp2p/specs/pull/550
+
+## Specification
+
+### GET `/ipni/v1/ad/head`
+
+This endpoint retrieves the most recent advertisement CID published by the provider, along with additional information such as the provider's public key and the topic under which the advertisement is published.
```

Comment:
> topic under which the advertisement is published

Should this say "topic under which the advertisement is announced"?  If this is saying "published" then that seems to imply that the topic would be used to construct the URL path to retrieve the advertisement, but that is not stated anywhere.